common/lib/host_list_provider/monitoring/cluster_topology_monitor.ts

```diff
@@ -0,0 +1,517 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ 
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+ 
+  http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { HostInfo } from "../../host_info";
+import { CacheMap } from "../../utils/cache_map";
+import { PluginService } from "../../plugin_service";
+import { HostAvailability } from "../../host_availability/host_availability";
+import { logTopology, sleep } from "../../utils/utils";
+import { logger } from "../../../logutils";
+import { HostRole } from "../../host_role";
+import { ClientWrapper } from "../../client_wrapper";
+import { AwsWrapperError } from "../../utils/errors";
+import { MonitoringRdsHostListProvider } from "./monitoring_host_list_provider";
+import { Messages } from "../../utils/messages";
+
+export interface ClusterTopologyMonitor {
+  forceRefresh(client: any, timeoutMs: number): Promise<HostInfo[]>;
```

Comment:
```suggestion
  forceRefresh(client: ClientWrapper, timeoutMs: number): Promise<HostInfo[]>;
```

Avoid using `any` for `client`